spring-aop/src/main/java/org/springframework/aop/framework/AdvisorChainFactory.java

```diff
@@ -27,6 +27,7 @@
  * @author Rod Johnson
  * @author Juergen Hoeller
  */
+@FunctionalInterface
```

Comment:
In my initial pass adding `@FunctionalInterface` a while ago, I did indeed skip interfaces which just happen to declare a single method for other reasons. From my perspective, a single abstract method is a required but not sufficient condition for calling an interface "functional".